core/src/test/java/org/elasticsearch/search/highlight/HighlightBuilderTests.java

```diff
@@ -287,44 +292,73 @@ public MappedFieldType fieldMapper(String name) {
-            assertEquals(highlight.globalForceSource(), parsedHighlight.globalForceSource());
-            assertEquals(highlight.fields().size(), parsedHighlight.fields().size());
-
-            Iterator<org.elasticsearch.search.highlight.SearchContextHighlight.Field> iterator = parsedHighlight.fields().iterator();
             for (org.elasticsearch.search.highlight.SearchContextHighlight.Field field : highlight.fields()) {
-                org.elasticsearch.search.highlight.SearchContextHighlight.Field otherField = iterator.next();
-                assertEquals(field.field(), otherField.field());
+                String encoder = highlightBuilder.encoder() != null ? highlightBuilder.encoder() : HighlightBuilder.DEFAULT_ENCODER;
+                assertEquals(encoder, field.fieldOptions().encoder());
+                HighlightBuilder.Field tmp = null;
+                for (HighlightBuilder.Field hbfield : highlightBuilder.fields()) {
+                    if (hbfield.name().equals(field.field())) {
+                        tmp = hbfield;
+                    }
+                }
+                final HighlightBuilder.Field fieldBuilder = tmp;
                 FieldOptions options = field.fieldOptions();
-                FieldOptions otherOptions = otherField.fieldOptions();
-                assertArrayEquals(options.boundaryChars(), options.boundaryChars());
-                assertEquals(options.boundaryMaxScan(), otherOptions.boundaryMaxScan());
-                assertEquals(options.encoder(), otherOptions.encoder());
-                assertEquals(options.fragmentCharSize(), otherOptions.fragmentCharSize());
-                assertEquals(options.fragmenter(), otherOptions.fragmenter());
-                assertEquals(options.fragmentOffset(), otherOptions.fragmentOffset());
-                assertEquals(options.highlighterType(), otherOptions.highlighterType());
-                assertEquals(options.highlightFilter(), otherOptions.highlightFilter());
-                assertEquals(options.highlightQuery(), otherOptions.highlightQuery());
-                assertEquals(options.matchedFields(), otherOptions.matchedFields());
-                assertEquals(options.noMatchSize(), otherOptions.noMatchSize());
-                assertEquals(options.numberOfFragments(), otherOptions.numberOfFragments());
-                assertEquals(options.options(), otherOptions.options());
-                assertEquals(options.phraseLimit(), otherOptions.phraseLimit());
-                assertArrayEquals(options.preTags(), otherOptions.preTags());
-                assertArrayEquals(options.postTags(), otherOptions.postTags());
-                assertEquals(options.requireFieldMatch(), otherOptions.requireFieldMatch());
-                assertEquals(options.scoreOrdered(), otherOptions.scoreOrdered());
+                BiConsumer<Function<AbstractHighlighterBuilder<?>, Object>, Function<FieldOptions, Object>> checkSame = (accessor1,
```

Comment:
This is cute! It is worth commenting and probably it is worth naming `accessor1` and `accessor2` something more descriptive. I might make a static method that builds the BiConsumer so that I have a convenient place to put the comment.